test/functional/terminal/tui_spec.lua

```diff
@@ -438,34 +438,29 @@ describe('TUI FocusGained/FocusLost', function()
     feed_data(':set shell='..nvim_dir..'/shell-test\n')
     feed_data(':set noshowmode laststatus=0\n')
 
-    retry(3, nil, function()
-      feed_data(':terminal\n')
-      screen:sleep(1)
-      feed_data('\027[I')
-      screen:expect([[
-        {1:r}eady $                                           |
-        [Process exited 0]                                |
-                                                          |
-                                                          |
-                                                          |
-        gained                                            |
-        {3:-- TERMINAL --}                                    |
-      ]])
-      feed_data('\027[O')
-      screen:expect([[
-        {1:r}eady $                                           |
-        [Process exited 0]                                |
-                                                          |
-                                                          |
-                                                          |
-        lost                                              |
-        {3:-- TERMINAL --}                                    |
-      ]])
-
-      -- If retry is needed...
-      feed_data("\034\016")  -- CTRL-\ CTRL-N
-      feed_data(':bwipeout!\n')
-    end)
+    feed_data(':terminal\n')
+    screen:sleep(1)
+    feed_data('\027[I')
+    screen:expect({grid=[[
+      {1:r}eady $                                           |
+      [Process exited 0]                                |
+                                                        |
+                                                        |
+                                                        |
+      gained                                            |
+      {3:-- TERMINAL --}                                    |
+    ]], timeout=(3 * screen.timeout)})
```

Comment:
I don't think this will help. The reason for the retry() was because of potential "racy" behavior with `feed_data('\027[I')` or focus gained/lost ordering.

Reply:
Ordering at least should not be a problem when doing this only once, and waiting for the first event.

Reply:
we can try it

Reply:
ok, seems to be stable so far.
Triggered another rebuild.